src/ripple/basics/TaggedCache.h

```diff
@@ -196,14 +196,17 @@ class TaggedCache
         return true;
     }
 
+    using SweptPointersVector = std::pair<
+        std::vector<std::shared_ptr<mapped_type>>,
+        std::vector<std::weak_ptr<mapped_type>>>;
+
     void
     sweep()
     {
         // Keep references to all the stuff we sweep
         // For performance, each worker thread should exit before the swept data
         // is destroyed but still within the main cache lock.
-        std::vector<std::vector<std::shared_ptr<mapped_type>>> allStuffToSweep(
-            m_cache.partitions());
+        std::vector<SweptPointersVector> allStuffToSweep(m_cache.partitions());
```

Comment:
I don't think it would work, because partitions are processed concurrently in different threads, so then need different vectors. I don't believe the `reserve()` in `sweepHelper` is done under a lock that protects from concurrent access from the workers.

Reply:
Oh *duh*. Of course this would be a terrible idea with multiple threads. Nevermind! 😃